controllers/warehouse-controller.js

```diff
@@ -174,10 +174,68 @@ const editWarehouseById = async (req, res) => {
   }
 }
 
+//POST a new warehouse item
+const postNewWarehouseItem = async (req, res) => {
+  try {
+    const {
+      warehouse_name,
+      address,
+      city,
+      country,
+      contact_name,
+      contact_position,
+      contact_phone,
+      contact_email,
+    } = req.body;
+
+    // Validate data
+    const validation = await validateNewWarehouseItem(req.body);
+    if (!validation.isValid) {
+      console.log({ message: validation.message });
+      return res.status(400).json({ message: validation.message });
+    }
+
+    // Insert new warehouse item into the database
+    const [newItemId] = await knex("warehouses").insert({
+      warehouse_name,
+      address,
+      city,
+      country,
+      contact_name,
+      contact_position,
+      contact_phone,
+      contact_email,
+    });
+
+    // Retrieve and respond with the inserted item
```

Comment:
@Maedehshariifii I think this is redundant, we don't need to retrieve new item we just added, id validation worked, we could use the request body as the return

Reply:
Agree. Just check if the write to DB was successful and then just return that body.